src/scalar.rs

```diff
@@ -465,18 +478,47 @@ impl PrimeField for Scalar {
     }
 }
 
+#[cfg(not(target_pointer_width = "64"))]
+type ReprBits = [u32; 8];
+
+#[cfg(target_pointer_width = "64")]
+type ReprBits = [u64; 4];
+
 impl PrimeFieldBits for Scalar {
     // Representation in non-Montgomery form.
-    type ReprBits = [u64; 4];
```

Comment:
What prevents you from `type ReprBits = [u8; 32];` and then
```
impl PrimeFieldBits for Scalar {
    // Representation in non-Montgomery form.
    type ReprBits = [u8; 32];

    fn to_le_bits(&self) -> FieldBits<Self::ReprBits> {
        let mut scalar = blst_scalar::default();
        unsafe { blst_scalar_from_fr(&mut scalar, &self.0) };

        FieldBits::new(scalar.b)
    }

    fn char_le_bits() -> FieldBits<Self::ReprBits> {
        FieldBits::new(MODULUS_REPR)
    }
}
```
In other words without dependency on `target_pointer_width`.

Reply:
Not exactly related to this, but I've spotted
```
    /// Converts an element of `Scalar` into a byte representation in
    /// little-endian byte order.
    #[inline]
    pub fn to_bytes_le(&self) -> [u8; 32] {
        let mut out = [0u64; 4];
        unsafe { blst_uint64_from_fr(out.as_mut_ptr(), &self.0) };
        out.as_byte_slice().try_into().unwrap()
    }
```
This can also be rewritten using blst_scalar_from_fr. This way you'll remove dependency on endiannes. [At least in this case of course.]

Reply:
> What prevents you from type ReprBits = [u8; 32]; and then

The point of this representation is to be `u64` or `u32`s due to how it is used

Reply:
The `FieldBits` accepts a range of types, `u8` included, and `blst_scalar` is in right byte order. Note that `cargo test` passes...